### PR TITLE
Feature/suitep dv user profile

### PR DESCRIPTION
**Pull request by daniel-samson**
## Description

In relation to issue #2240, this change modifies the user profile detail view so that it is in keeping with SuiteP while being in keeping with the older themes.
- added the advanced and access tab back into suite. 
- added the reset user profile features back into suite.
- added a generic modal dialog to suite (bootstrap).
- refactored the detailview
- refactored related javascript.
- moves the all the view code into detailview.tpl
- changed detail view defs
- moved roles subpanel to the subpanels instead of in the access tab.
- made employee info panel collapsed 

user-profile-detail-view

It also fixes issue #2540.

## Motivation and Context

Adds missing features back into suite.
## How To Test This
* Rebuild style.css from sass

Access the users detail view.
## Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [x] Breaking change (fix or feature that would cause existing functionality to change)
### Final checklist

- [x] My code follows the code style of this project found [here](https://suitecrm.com/wiki/index.php/Coding_Standards).
- [ ] My change requires a change to the documentation.
- [x] I have read the [**How to Contribute**](https://suitecrm.com/wiki/index.php/Contributing_to_SuiteCRM) guidelines.

### CODE REVIEW

File: modules/ACLRoles/DetailUserAccess.php [hunk 1]
```diff
@@ -0,0 +1,95 @@
+<?php
+if(!defined('sugarEntry') || !sugarEntry) die('Not A Valid Entry Point');
+/*********************************************************************************
+ * SugarCRM Community Edition is a customer relationship management program developed by
+ * SugarCRM, Inc. Copyright (C) 2004-2013 SugarCRM Inc.
+
+ * SuiteCRM is an extension to SugarCRM Community Edition developed by Salesagility Ltd.
+ * Copyright (C) 2011 - 2014 Salesagility Ltd.
```
Comment (samus-aran) [inline modules/ACLRoles/DetailUserAccess.php]: Update License

File: modules/ACLRoles/DetailUserAccess.php [hunk 2]
```diff
@@ -0,0 +1,95 @@
...
+    $title = getClassicModuleTitle( '',array($mod_strings['LBL_MODULE_NAME'],$mod_strings['LBL_ROLES_SUBPANEL_TITLE']), '');
+    
+    $sugar_smarty->assign('TITLE', $title);
+    $sugar_smarty->assign('USER_ID', $focus->id);
+    $sugar_smarty->assign('LAYOUT_DEF_KEY', 'UserRoles');
+    echo $sugar_smarty->fetch('modules/ACLRoles/DetailViewUser.tpl');
+    
+    
+    //this gets its layout_defs.php file from the user not from ACLRoles so look in modules/Users for the layout defs
+//    require_once('include/SubPanel/SubPanelTiles.php');
```
Comment (samus-aran) [inline modules/ACLRoles/DetailUserAccess.php]: Remove, not comment out

File: modules/Users/DetailView.js [hunk 3]
```diff
@@ -1,6 +1,9 @@
 /*********************************************************************************
  * SugarCRM Community Edition is a customer relationship management program developed by
  * SugarCRM, Inc. Copyright (C) 2004-2013 SugarCRM Inc.
+
+ * SuiteCRM is an extension to SugarCRM Community Edition developed by Salesagility Ltd.
+ * Copyright (C) 2011 - 2014 Salesagility Ltd.
```
Comment (samus-aran) [inline modules/Users/DetailView.js]: Update License

File: modules/Users/language/en_us.lang.php [hunk 4]
```diff
@@ -94,11 +94,11 @@
     'LBL_ADDRESS_CITY' => 'Address City',
```
Comment (samus-aran) [inline modules/Users/language/en_us.lang.php]: Update Formatting (supplied XML)

File: modules/Users/views/view.detail.php [hunk 5]
```diff
@@ -95,48 +100,53 @@ function preDisplay() {
...
+
         $this->ss->assign('SHOW_ROLES', $show_roles);
         //Mark whether or not the user is a group or portal user
         $this->ss->assign('IS_GROUP_OR_PORTAL', ($this->bean->is_group=='1' || $this->bean->portal_only=='1') ? true : false);
         if ( $show_roles ) {
             ob_start();
-            echo "<div>";
-            require_once('modules/ACLRoles/DetailUserRole.php');
-            echo "</div></div>";
+//            echo "<div>";
```
Comment (samus-aran) [inline modules/Users/views/view.detail.php]: Remove //

File: themes/SuiteP/modules/Users/tpls/DetailView.tpl [hunk 6]
```diff
@@ -0,0 +1,264 @@
+{*
+/**
+ *
+ * SugarCRM Community Edition is a customer relationship management program developed by
+ * SugarCRM, Inc. Copyright (C) 2004-2013 SugarCRM Inc.
+ *
+ * SuiteCRM is an extension to SugarCRM Community Edition developed by SalesAgility Ltd.
+ * Copyright (C) 2011 - 2016 SalesAgility Ltd.
```
Comment (samus-aran) [inline themes/SuiteP/modules/Users/tpls/DetailView.tpl]: Update License (year)

**Comment by daniel-samson** [comment]
I have updated with the changes you have requested

File: modules/Users/DetailView.js (hunk 3 above)
Comment (mattlorimer) [inline modules/Users/DetailView.js]: still the wrong License

File: modules/Users/language/en_us.lang.php [hunk 7]
```diff
@@ -1,5 +1,7 @@
 <?php
-if (!defined('sugarEntry') || !sugarEntry) die('Not A Valid Entry Point');
+if (!defined('sugarEntry') || !sugarEntry) {
+    die('Not A Valid Entry Point');
+}
 /*********************************************************************************
```
Comment (mattlorimer) [inline modules/Users/language/en_us.lang.php]: old style Licence

File: modules/Users/language/en_us.lang.php [hunk 8]
```diff
@@ -1,5 +1,7 @@
 <?php
-if (!defined('sugarEntry') || !sugarEntry) die('Not A Valid Entry Point');
+if (!defined('sugarEntry') || !sugarEntry) {
```
Comment (mattlorimer) [inline modules/Users/language/en_us.lang.php]: move below Licence

File: modules/Users/metadata/detailviewdefs.php [hunk 9]
```diff
@@ -37,40 +37,140 @@
  * display the words  "Powered by SugarCRM" and "Supercharged by SuiteCRM".
  ********************************************************************************/
```
Comment (mattlorimer) [inline modules/Users/metadata/detailviewdefs.php]: update licence

File: modules/Users/tpls/DetailView.tpl [hunk 10]
```diff
@@ -0,0 +1,556 @@
+{*
+/*********************************************************************************
```
Comment (mattlorimer) [inline modules/Users/tpls/DetailView.tpl]: update Licence

File: include/DetailView/DetailView2.php [hunk 11]
```diff
@@ -48,7 +48,7 @@
 class DetailView2 extends EditView
 {
     var $view = 'DetailView';
```
Comment (mattlorimer) [inline include/DetailView/DetailView2.php]: update var to public

Comment (daniel-samson) [inline include/DetailView/DetailView2.php]: Changed

File: modules/Users/tpls/DetailView.tpl (hunk 10 above)
Comment (daniel-samson) [inline modules/Users/tpls/DetailView.tpl]: fixed

File: modules/Users/metadata/detailviewdefs.php (hunk 9 above)
Comment (daniel-samson) [inline modules/Users/metadata/detailviewdefs.php]: fixed

File: modules/Users/language/en_us.lang.php (hunk 7 above)
Comment (daniel-samson) [inline modules/Users/language/en_us.lang.php]: fixed

File: modules/Users/language/en_us.lang.php (hunk 8 above)
Comment (daniel-samson) [inline modules/Users/language/en_us.lang.php]: fixed

**Comment by samus-aran** [comment]
Looks Good and the passed Matt's Review (says I ;))

**Pull request merged**